processor/deltatocumulativeprocessor/factory.go

```diff
@@ -32,6 +33,13 @@ func createMetricsProcessor(_ context.Context, set processor.Settings, cfg compo
 	if err != nil {
 		return nil, err
 	}
+	proc := newProcessor(pcfg, set.Logger, telb, next)
 
-	return newProcessor(pcfg, set.Logger, telb, next), nil
+	ltel, err := ltel.New(set.TelemetrySettings)
+	if err != nil {
+		return nil, err
+	}
+	linear := newLinear(pcfg, ltel, proc)
+
+	return Chain{linear, proc}, nil
```

Comment:
Do we still need to chain them? What isn't yet implemented in Linear? IMO it would be much simpler (for metrics, file structure, etc) to just switch wholesale, rather than trying to keep both around.

Reply:
Linear only does sums on this branch.

Making linear do everything involves some fairly advanced generics usage, which I think deserves to be reviewed properly and probably separately :)

Given the code already exists on the non-partial branch, I expect to send the next patch right after merging this one.

We should take care to merge so that we only release after merging both.